### PR TITLE
feat: Add Gecko driver options

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Change list

https://github.com/appium/appium-geckodriver
 
## Types of changes

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 1]
```diff
@@ -129,16 +129,39 @@ public AppiumDriver(Capabilities capabilities) {
...
+            Capabilities originalCapabilities, String defaultName) {
+        String currentAutomationName = CapabilityHelpers.getCapability(
+                originalCapabilities, MobileCapabilityType.AUTOMATION_NAME, String.class);
+        if (isBlank(currentAutomationName)) {
+            ImmutableCapabilities toMerge = originalCapabilities.getCapabilityNames()
+                    .contains(MobileCapabilityType.AUTOMATION_NAME)
+                    ? new ImmutableCapabilities(MobileCapabilityType.AUTOMATION_NAME, defaultName)
+                    : new ImmutableCapabilities(
+                            APPIUM_PREFIX + MobileCapabilityType.AUTOMATION_NAME, defaultName);
+            return originalCapabilities.merge(toMerge);
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/AppiumDriver.java:160]:
```suggestion
            String capabilityName = originalCapabilities.getCapabilityNames().contains(
                    MobileCapabilityType.AUTOMATION_NAME) 
                    ? MobileCapabilityType.AUTOMATION_NAME
                    : APPIUM_PREFIX + MobileCapabilityType.AUTOMATION_NAME;
            ImmutableCapabilities toMerge = new ImmutableCapabilities(capabilityName, defaultName);
            return originalCapabilities.merge(toMerge);
```

File: src/main/java/io/appium/java_client/remote/options/SupportsBrowserVersionOption.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Provide the version number of the browser to automate if there are multiple
+     * versions installed on the same machine where the driver is running.
+     *
+     * @param version Browser version to use.
+     * @return self instance for chaining.
+     */
+    default T setBrowserVersion(String version) {
+        return amend(BROWSER_VERSION_OPTION, version);
+    }
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/remote/options/SupportsBrowserVersionOption.java:36]: no getter here?

Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/remote/options/SupportsBrowserVersionOption.java:36]: no need. The Capabilities interface already has it

File: src/main/java/io/appium/java_client/AppiumDriver.java (hunk 1 above)
Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/AppiumDriver.java]: changed

**Review by valfirst — APPROVED** [review]

**Pull request merged**